server/vulnerabilities/msrc/parsed/product.go

```diff
@@ -66,7 +66,53 @@ func (p Products) GetMatchForOS(ctx context.Context, os fleet.OperatingSystem) (
 }
 
 func NewProductFromFullName(fullName string) Product {
-	return Product(fullName)
+	// If the full name includes a version, return it as-is.
+	p := Product(fullName)
+	if p.HasDisplayVersion() {
+		return p
+	}
```

Comment:
This ensures this function is still idempotent and won't keep adding version strings.